c10/core/Device.h

```diff
@@ -86,6 +86,23 @@ struct C10_API Device final {
     return type_ == DeviceType::CPU;
   }
 
+  /// Return the right-biased merge of two devices.  This usually just returns
+  /// the right argument, except if self.type() == device.type() and
+  /// !device.has_index(), in which case we retain the index from self.  This
+  /// retains consistency with the new_* APIs from PyTorch, which also
+  /// respect original device indices in this way:
+  ///
+  ///     >>> torch.zeros(0, device='cuda:1').new_empty(1, device='cuda')
+  ///     tensor([0.], device='cuda:1')
+  ///
+  Device merge_in(Device device) const noexcept {
```

Comment:
you don't need this anymore?

Reply:
Not for this patch, but it seems like it might be useful for someone else, sometime?

Reply:
nope, I don't think it's useful.  The comment and analysis is all about not setting the device_guard properly.